### PR TITLE
Adds image-to-json translator.

### LINKED ISSUE #47: Add image-to-json translator.
Translates an image object into the various JSON objects (manifest/container configuration).

### PR DESCRIPTION
**Pull request by coollog**
Fixes #47

### CODE REVIEW

File: crepecake/src/main/java/com/google/cloud/tools/crepecake/json/ImageToJsonTranslator.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.json;
+
+import com.google.cloud.tools.crepecake.blob.Blob;
+import com.google.cloud.tools.crepecake.blob.BlobDescriptor;
+import com.google.cloud.tools.crepecake.image.DescriptorDigest;
+import com.google.cloud.tools.crepecake.image.Image;
```
Comment (elharo) [inline crepecake/src/main/java/com/google/cloud/tools/crepecake/json/ImageToJsonTranslator.java]: Something I just noted (not to be addressed in this PR). Is an Image always a Docker image? Can it be some other kind of image? If not, we might want to rename this class DockerImage.

File: crepecake/src/test/java/com/google/cloud/tools/crepecake/json/ImageToJsonTranslatorTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  @Test
+  public void testGetContainerConfiguration()
+      throws IOException, LayerPropertyNotFoundException, URISyntaxException {
+    // Loads the expected JSON string.
+    File jsonFile =
+        new File(getClass().getClassLoader().getResource("json/containerconfig.json").toURI());
+    final String expectedJson =
+        CharStreams.toString(new InputStreamReader(new FileInputStream(jsonFile)));
```
Comment (elharo) [inline crepecake/src/test/java/com/google/cloud/tools/crepecake/json/ImageToJsonTranslatorTest.java]: character set

File: crepecake/src/test/java/com/google/cloud/tools/crepecake/json/ImageToJsonTranslatorTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  }
+
+  @Test
+  public void testGetManifest()
+      throws URISyntaxException, IOException, LayerPropertyNotFoundException {
+    // Loads the expected JSON string.
+    File jsonFile =
+        new File(getClass().getClassLoader().getResource("json/translatedmanifest.json").toURI());
+    final String expectedJson =
+        CharStreams.toString(new InputStreamReader(new FileInputStream(jsonFile)));
```
Comment (elharo) [inline crepecake/src/test/java/com/google/cloud/tools/crepecake/json/ImageToJsonTranslatorTest.java]: character set

File: crepecake/src/main/java/com/google/cloud/tools/crepecake/json/ImageToJsonTranslator.java (hunk 1 above)
Comment (coollog) [reply crepecake/src/main/java/com/google/cloud/tools/crepecake/json/ImageToJsonTranslator.java]: The image can be either a Docker image or an OCI image. But in the first implementation, we will only be building a Docker image. Adding OCI image support should be fairly easy by changing some constants in the JSON templates.

**Pull request merged**